### PR TITLE
Build NuGet packages for Orleans 2.0 (.NET Standard)

### PR DESCRIPTION
**Pull request by jdom**
This builds the packages for the vNext solution

### CODE REVIEW

**Comment by jdom** [comment]
For the scripts themselves, it's easier to review individual commits to see what changedm since the first commit just copied a bunch of files with no changes from the main tree into vNext, so you can ignore reviewing it. The other commits are the interesting ones.

File: vNext/src/NuGet/Microsoft.Orleans.Core.nuspec [hunk 1]
```diff
@@ -0,0 +1,50 @@
+﻿<?xml version="1.0"?>
+<package xmlns="http://schemas.microsoft.com/packaging/2010/07/nuspec.xsd">
+  <metadata>
+    <id>Microsoft.Orleans.Core</id>
```
Comment (ReubenBond) [inline vNext/src/NuGet/Microsoft.Orleans.Core.nuspec]: I thought we were leaning towards calling this one "Abstractions"

File: vNext/src/NuGet/Microsoft.Orleans.Core.nuspec [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+    <version>$version$</version>
+    <title>Microsoft Orleans Core Library</title>
+    <authors>Microsoft</authors>
+    <owners>Microsoft,Orleans</owners>
+    <projectUrl>https://github.com/dotnet/Orleans</projectUrl>
+    <licenseUrl>https://github.com/dotnet/Orleans#license</licenseUrl>
+    <iconUrl>https://raw.githubusercontent.com/dotnet/orleans/gh-pages/assets/logo_128.png</iconUrl>
+    <summary>
+      Core library of Microsoft Orleans - Orleans.dll
```
Comment (ReubenBond) [inline vNext/src/NuGet/Microsoft.Orleans.Core.nuspec]: I think we should use a namespaced dll name and not include it in the package description

File: vNext/src/NuGet/Microsoft.Orleans.OrleansCodeGenerator.Build.nuspec [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+      </group>
+    </dependencies>
+  </metadata>
+  <files>
+    <!-- Auto-import MSBuild targets and props files into target project.
+           http://docs.nuget.org/docs/creating-packages/creating-and-publishing-a-package#Import_MSBuild_targets_and_props_files_into_project_(Requires_NuGet_2.5_or_above)
+      -->
+    <file src="Orleans.SDK.targets" target="build\Microsoft.Orleans.OrleansCodeGenerator.Build.targets" />
+        
+    <file src="ClientGenerator.exe" target="tools" />
```
Comment (ReubenBond) [inline vNext/src/NuGet/Microsoft.Orleans.OrleansCodeGenerator.Build.nuspec]: Shouldn't these be included via dependencies on other packages?

File: vNext/src/NuGet/Microsoft.Orleans.Core.nuspec (hunk 2 above)
Comment (jdom) [inline vNext/src/NuGet/Microsoft.Orleans.Core.nuspec]: Agreed. Nevertheless those changes are not part of this PR, that's why I was mentioning you should skip reviewing the first commit, as it is a plain copy of what's out there already.

File: vNext/src/NuGet/Microsoft.Orleans.OrleansCodeGenerator.Build.nuspec (hunk 3 above)
Comment (jdom) [inline vNext/src/NuGet/Microsoft.Orleans.OrleansCodeGenerator.Build.nuspec]: Not sure if it's possible, since this is really a tool package. Nevertheless, I only copied how we were doing it before, and I believe after the .NET Core tooling settles and starts using MSBuild again, we should improve this story

**Comment by galvesribeiro** [comment]
Yay! Lets test Nugets! :) 🔥

**Comment by jdom** [comment]
I'm doing that here: https://github.com/jdom/OrleansHelloWorldSample.Core
There is one thing it's not working yet, but with nuget packages being generated, now it's easy to iterate :)
Don't worry, I'll have it working really soon

**Comment by jdom** [comment]
YEAH! Got a HelloWorld sample to work ENTIRELY in .NET Core. I had to hardcode a few values in our AppDomain shim, but I'll clean in up on Monday and submit a PR
Exciting! 🎉

**Comment by galvesribeiro** [comment]
Great work! 💃

**Comment by jskulavik** [comment]
Still getting up to speed here, but that's fantastic (and very useful to me) news @jdom.  Bravo!
